lib/Sema/CSSimplify.cpp

```diff
@@ -8806,8 +8908,16 @@ ConstraintSystem::simplifyApplicableFnConstraint(
   // is not valid for operators though, where an inout parameter does not
   // have an explicit inout argument.
   if (type1.getPointer() == desugar2) {
-    if (!isOperator || !hasInOut())
+    if (!isOperator || !hasInOut()) {
+      if (locator.isForImplicitCallOfCallableValue()) {
+        // This overload can be trivially called to produce a value of expected
+        // type record the insert explicit call fix.
```

Comment:
Sorry, I don't understand this comment. Could you please reword?

Reply:
It was indeed confusing, fixed!